mockup_package/mockup/image.py

```diff
@@ -43,6 +43,48 @@ async def mockup(location, device_id, original_img_path_list, device_info):
     return output_img_path_list
 
 
+# Now only the first orientation model is generated for preview
+async def previewMockup(
+    location, device_id, original_img_path_list, device_info, preview_img_index=0
+):
```

Comment:
1. should we support `preview_img_index: List[int] = [0]` for future?
2. should we add types in python from now on? seems more readable

```python
async def previewMockup(
preview_img_index: List[int] = [0]
) -> List[someType]: # or Tuple better?
```

> Starting with Python 3.6, you can declare types of variables and functions, like this :

> `explicit_number: type`

Reply:
@pkong-ds I think we won't preview more than one orientation at once? if the user change orientation or upload new image, should generate a new preview image for it
and for types in python, I think yes, will add the type if I find the type missing

Reply:
Which UX do you think is better? After user upload image
1. orientiationA loading -> show oriA preview -> click oriB -> _oriB loading_ -> show oriB preview
2. allOri loading -> show oriA preview -> click oriB -> show oriB preview

Or even better
3. Can we parallel this somehow, so that generation of oriA would not block oriB?

Reply:
> if the user change orientation or upload new image, should generate a new preview image for it
btw, according to this logic, `original_img_path_list: List[str]` can just be `original_img_path: str`?

Reply:
@pkong-ds I think 1 is better because case 2 the user needs to wait longer,, but we can store the result once orientiationA generated and don't need to generate again if user change back to select orientiationA.

`original_img_path_list: List[str]` is for the uploaded images, and `preview_image_index` is for which orientation use for preview, let me update `preview_image_index` to `preview_orientation_index` to avoid confusion

Reply:
Thanks a lot. 

I was saying original_path because we wont process multiple images here? It's always a List of ONE single item?

That's why I suggested changing it as a str variable, over List[str]

Reply:
@pkong-ds oh, as the logic is almost same, I update to merge `mockup` and `previewMockup` to one function now, and will generate all mockup when `preview_orientation_index = None`

Reply:
@YayunHuang I disagree with merging 2 as one function.

1. they serve 2 purposes, as you mentioned, `mockup` & `previewMockup`
2. IMO `their logic being same now` != `we should merge the f unction`. Imagine if future, changing features about `previewMockup` will change this shared function, which might leads to future devs misunderstanding such changes as related to `mockup`
3. `original_img_path` list is a good example of incorrect abstraction already.
    - `mockup` supports list of images
    - `previewMockup` always expects single image
4. You can extract the `generate` part as a separate function

```js
function generate() {  
  // ... duplicate logic
}

function mockup {  
  generate()
}

function previewMockup {
  generate()
}

```